File: scripts/encode-run-transfer.mjs
import fs from 'node:fs';
import {spawnSync} from 'node:child_process';
import {createHash} from 'node:crypto';
import assert from 'node:assert/strict';
const ffmpeg=process.argv[2];if(!ffmpeg)throw Error('Pass the installed ffmpeg executable path');
const dir='dist-work/run-transfer-20260913',report=[];
function run(args){const r=spawnSync(ffmpeg,args,{encoding:'utf8',windowsHide:true,maxBuffer:8e6});assert.equal(r.status,0,r.stderr);return r}
for(const key of ['jog','sprint']){
 const file=`SwarmFront-${key}-vs-current-20260913.mp4`,path=dir+'/'+file;
 assert.equal(fs.readdirSync(`${dir}/${key}-frames`).filter(p=>p.endsWith('.jpg')).length,360);
 run(['-y','-framerate','30','-i',`${dir}/${key}-frames/%04d.jpg`,'-frames:v','360','-c:v','libx264','-preset','veryfast','-crf','19','-pix_fmt','yuv420p','-movflags','+faststart','-an',path]);
 const decode=run(['-i',path,'-progress','pipe:1','-f','null','-']);assert.match(decode.stdout,/frame=360\s/);assert.match(decode.stderr,/1280x720/);assert.match(decode.stderr,/30 fps/);
 const frames=[12,24,36,48,132,144,156,168,252,264,276,288];
 run(['-y','-i',path,'-vf',`select='${frames.map(n=>`eq(n,${n})`).join('+')}',scale=320:180,tile=4x3`,'-frames:v','1',`${dir}/${key}-video-contact.jpg`]);
 report.push({key,file,bytes:fs.statSync(path).size,sha256:createHash('sha256').update(fs.readFileSync(path)).digest('hex'),fps:30,frames:360,seconds:12,codec:'H264 yuv420p',width:1280,height:720,allFramesDecoded:true,contactSheetFrames:frames});console.log('Encoded and decoded',key,fs.statSync(path).size,'bytes');
}
fs.writeFileSync(dir+'/video-verification.json',JSON.stringify(report,null,2));

File: scripts/encode-run-transfer.py
import bpy,json
from pathlib import Path
OUT=Path(__file__).resolve().parents[1]/'dist-work/run-transfer-20260913'
scene=bpy.context.scene;scene.render.fps=30;scene.render.resolution_x=1280;scene.render.resolution_y=720;scene.render.resolution_percentage=100
scene.view_settings.view_transform='Standard';scene.view_settings.look='None';scene.sequencer_colorspace_settings.name='sRGB'
scene.render.use_sequencer=True;scene.render.use_compositing=False
editor=scene.sequence_editor_create()
for key in ['jog','sprint']:
 files=sorted((OUT/(key+'-frames')).glob('*.jpg'));assert len(files)==360
 strip=editor.strips.new_image(key,str(files[0]),channel=1,frame_start=1)
 for f in files[1:]:strip.elements.append(f.name)
 strip.frame_final_duration=len(files);scene.frame_start=1;scene.frame_end=len(files)
 scene.render.image_settings.media_type='VIDEO';scene.render.ffmpeg.format='MPEG4';scene.render.ffmpeg.codec='H264';scene.render.ffmpeg.constant_rate_factor='MEDIUM';scene.render.ffmpeg.ffmpeg_preset='GOOD';scene.render.filepath=str(OUT/('SwarmFront-'+key+'-vs-current-20260913.mp4'))
 bpy.ops.render.render(animation=True);editor.strips.remove(strip)
print('ENCODED 2 x 12 seconds, 1280x720 H264 30fps')
